Feature/communication

maydek1 opened the pull request.

maydek1 committed:
feat(passenger): integrate RabbitMQ and Feign clients, remove PassengerService interface

- Added RabbitMQ to docker-compose for message-based communication
- Implemented new DTOs in Passenger service
- Configured RabbitMQ and established async communication with Driver service
- Added and configured FeignClient for integration with Rating and Ride services
- Removed PassengerService interface to simplify the service layer

maydek1 committed:
feat(driver): integrate Feign client, remove DriverService interface

- Implemented new DTOs in Driver service
- Configured RabbitMQ and established async communication with Passenger service
- Added and configured FeignClient for integration with Ride service
- Removed DriverService interface to simplify the service layer

maydek1 committed:
refactor(car): improve code readability

Naza1re commented on passenger-service/src/main/java/com/software/modsen/passengerservice/client/RatingClient.java:
```diff
@@ -0,0 +1,15 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@FeignClient(name = "rating-service", url = "http://localhost:4569/ratings", configuration = FeignConfig.class)
```
url и name вынести в application.yaml, и от туда парсить сюда

maydek1 replied:
Вынес

Naza1re commented on passenger-service/src/main/java/com/software/modsen/passengerservice/client/RideClient.java:
```diff
@@ -0,0 +1,27 @@
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@FeignClient(name = "ride-service", url = "http://localhost:4568/ride", configuration = FeignConfig.class)
```
тоже самое что и выше

Naza1re commented on passenger-service/src/main/java/com/software/modsen/passengerservice/client/RideClient.java:
```diff
@@ -0,0 +1,27 @@
+@FeignClient(name = "ride-service", url = "http://localhost:4568/ride", configuration = FeignConfig.class)
+public interface RideClient {
+
+    @PatchMapping("/status/")
```
лишний "/"

maydek1 replied:
Убрал

Naza1re commented on passenger-service/src/main/java/com/software/modsen/passengerservice/client/RideClient.java:
```diff
@@ -0,0 +1,27 @@
+    @PatchMapping("/status/")
+    ResponseEntity<RideResponse> changeStatus(@RequestBody RideStatusRequest rideStatusRequest);
+
+    @PostMapping()
```
лишние "()"

maydek1 replied:
Убрал

Naza1re commented on passenger-service/src/main/java/com/software/modsen/passengerservice/config/RabbitConfig.java:
```diff
@@ -0,0 +1,30 @@
+
+    @Bean
+    public Queue requestQueue() {
+        return new Queue("request.queue");
```
Вынести в application.yaml и парсить сюда

Naza1re commented on passenger-service/src/main/java/com/software/modsen/passengerservice/config/RabbitConfig.java:
```diff
@@ -0,0 +1,30 @@
+    @Bean
+    public RabbitTemplate rabbitTemplate(ConnectionFactory connectionFactory) {
+        RabbitTemplate template = new RabbitTemplate(connectionFactory);
+        template.setReplyTimeout(5000);
```
5000 лучше вынести в константу

maydek1 replied:
Вынес все в константы

Naza1re commented on passenger-service/src/main/java/com/software/modsen/passengerservice/config/RabbitConfig.java:
```diff
@@ -0,0 +1,30 @@
+
+    @Bean
+    public Queue responseQueue() {
+        return new Queue("response.queue");
```
тоже вынести в application.yaml

Naza1re commented on passenger-service/src/main/java/com/software/modsen/passengerservice/service/RideService.java:
```diff
@@ -0,0 +1,52 @@
+    private final Queue responseQueue;
+
+    public RideResponse startRide(RideRequest rideRequest) {
+        System.out.println(rideRequest);
```
Все System.out.println следует убрать из этого класса

maydek1 replied:
Убрал

Naza1re commented on driver-service/src/main/java/com/software/modsen/driverservice/client/RideClient.java:
```diff
@@ -0,0 +1,26 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+
+@FeignClient(name = "ride-service", url = "http://localhost:4568/ride", configuration = FeignConfig.class)
```
Внести в application.yaml

Naza1re commented on driver-service/src/main/java/com/software/modsen/driverservice/service/DriverListener.java:
```diff
@@ -0,0 +1,32 @@
+        responseProperties.setCorrelationId(correlationId);
+
+        Message responseMessage = new Message(driverId.toString().getBytes(), responseProperties);
+        System.out.println(responseMessage);
```
Лучше заменить выводы в консоль на логирование

maydek1 replied:
Убрал выводы

Naza1re commented on driver-service/src/main/java/com/software/modsen/driverservice/config/RabbitConfig.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Bean
+    public Queue requestQueue() {
+        return new Queue("request.queue");
```
Вынести все что возможно в application.yaml

Naza1re commented on driver-service/src/main/java/com/software/modsen/driverservice/service/DriverListener.java:
```diff
@@ -0,0 +1,32 @@
+    private final DriverRepository driverRepository;
+    private final RabbitTemplate rabbitTemplate;
+
+    @RabbitListener(queues = "request.queue")
```
Тоже вынести в applicatiom.yaml

Naza1re commented on driver-service/src/main/java/com/software/modsen/driverservice/service/DriverService.java:
```diff
@@ -1,14 +1,116 @@
+        return driverMapper.driverToDriverResponse(driverRepository.save(driver));
+    }
+
+    public int setRating(int ratingCount, int rate, int oldRate){
```
Рейтинг может быт не целым числом, возможно лучше использовать double в return

maydek1 replied:
Изменил

maydek1 committed:
refactor: move mapping to controller, update configs, remove CarService interface

- Moved DTO mapping logic to controller level
- Extracted configuration values to application.yml or constants
- Removed CarService interface for simplification
- Changed variable type for 'rate' in Driver entity
